🟠 Test failure on master

<!-- Please include a self-contained copy-pastable example that generates the issue if possible.

Please be concise with code posted. See guidelines below on how to provide a good bug report:

- Craft Minimal Bug Reports: http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports
- Minimal Complete Verifiable Examples: https://stackoverflow.com/help/mcve

Bug reports that follow these guidelines are easier to diagnose, and so are often handled much more quickly.
-->

**What happened**:

We have a failure related to a dask release, I think. Here's a job that failed: https://github.com/pydata/xarray/pull/5365/checks?check_run_id=2757459587

It's the test: `xarray/tests/test_computation.py::test_vectorize_dask_dtype_meta`

```

        References
        ----------
        .. [1] https://docs.scipy.org/doc/numpy/reference/ufuncs.html
        .. [2] https://docs.scipy.org/doc/numpy/reference/c-api/generalized-ufuncs.html
        """
        # Input processing:
        ## Signature
        if not isinstance(signature, str):
            raise TypeError("`signature` has to be of type string")
        input_coredimss, output_coredimss = _parse_gufunc_signature(signature)
    
        ## Determine nout: nout = None for functions of one direct return; nout = int for return tuples
        nout = None if not isinstance(output_coredimss, list) else len(output_coredimss)
    
        ## Consolidate onto `meta`
        if meta is not None and output_dtypes is not None:
>           raise ValueError(
                "Only one of `meta` and `output_dtypes` should be given (`meta` is preferred)."
            )
E           ValueError: Only one of `meta` and `output_dtypes` should be given (`meta` is preferred).

```

Should we xfail this? Does anyone have thoughts for a quick fix?

--- a/xarray/tests/test_computation.py
+++ b/xarray/tests/test_computation.py
@@ -1306,7 +1306,7 @@ def test_vectorize_dask_dtype_without_output_dtypes(data_array):
     assert expected.dtype == actual.dtype
 
 
-@pytest.mark.skip(
+@pytest.mark.skipif(
     LooseVersion(dask.__version__) > "2021.06",
     reason="dask/dask#7669: can no longer pass output_dtypes and meta",
 )

--- a/xarray/tests/test_computation.py
+++ b/xarray/tests/test_computation.py
@@ -1306,15 +1306,17 @@ def test_vectorize_dask_dtype_without_output_dtypes(data_array):
     assert expected.dtype == actual.dtype
 
 
-@pytest.mark.xfail(LooseVersion(dask.__version__) < "2.3", reason="dask GH5274")
+@pytest.mark.skip(
+    LooseVersion(dask.__version__) > "2021.06",
+    reason="dask/dask#7669: can no longer pass output_dtypes and meta",
+)
 @requires_dask
 def test_vectorize_dask_dtype_meta():
     # meta dtype takes precedence
     data_array = xr.DataArray([[0, 1, 2], [1, 2, 3]], dims=("x", "y"))
     expected = xr.DataArray([1, 2], dims=["x"])
 
-    func = functools.partial(
-        apply_ufunc,
+    actual = apply_ufunc(
         pandas_median,
         data_array.chunk({"x": 1}),
         input_core_dims=[["y"]],
@@ -1324,14 +1326,8 @@ def test_vectorize_dask_dtype_meta():
         dask_gufunc_kwargs=dict(meta=np.ndarray((0, 0), dtype=float)),
     )
 
-    # dask/dask#7669: can no longer pass output_dtypes and meta
-    if LooseVersion(dask.__version__) >= "2021.06":
-        with pytest.raises(ValueError):
-            func()
-    else:
-        actual = func()
-        assert_identical(expected, actual)
-        assert float == actual.dtype
+    assert_identical(expected, actual)
+    assert float == actual.dtype
 
 
 def pandas_median_add(x, y):

--- a/xarray/tests/test_computation.py
+++ b/xarray/tests/test_computation.py
@@ -1313,7 +1313,8 @@ def test_vectorize_dask_dtype_meta():
     data_array = xr.DataArray([[0, 1, 2], [1, 2, 3]], dims=("x", "y"))
     expected = xr.DataArray([1, 2], dims=["x"])
 
-    actual = apply_ufunc(
+    func = functools.partial(
+        apply_ufunc,
         pandas_median,
         data_array.chunk({"x": 1}),
         input_core_dims=[["y"]],
@@ -1323,8 +1324,14 @@ def test_vectorize_dask_dtype_meta():
         dask_gufunc_kwargs=dict(meta=np.ndarray((0, 0), dtype=float)),
     )
 
-    assert_identical(expected, actual)
-    assert float == actual.dtype
+    # dask/dask#7669: can no longer pass output_dtypes and meta
+    if LooseVersion(dask.__version__) >= "2021.06":
+        with pytest.raises(ValueError):
+            func()
+    else:
+        actual = func()
+        assert_identical(expected, actual)
+        assert float == actual.dtype
 
 
 def pandas_median_add(x, y):

